util/app/discovery/discovery.go

```diff
@@ -178,6 +179,7 @@ func cmpSupports(ctx context.Context, pluginSockFilePath, appPath, repoPath, fil
 		if namedPlugin {
 			return conn, cmpClient, true
 		}
+		io.Close(conn)
```

Comment:
Let's wrap connection closing in defer block.

Reply:
I don't think deferring buys us anything. I like immediately closing before returning nil.

Reply:
Yeah the problem is that we can't do one blanket `defer` since we don't know at the start whether we'll want to return the live connection. So whether we use `defer` or explicitly close, we have to do so in multiple early-return blocks.